File: Java-Core/src/main/java/presto/literal/DictEntryList.java
package presto.literal;

import java.util.LinkedList;

import presto.utils.CodeWriter;

public class DictEntryList extends LinkedList<DictEntry> {

	private static final long serialVersionUID = 1L;

	public DictEntryList() {
	}
	
	public DictEntryList(DictEntry entry) {
		this.add(entry);
	}
	
	@Override
	public String toString() {
		String text = super.toString();
		return '{' + text.substring(1,text.length()-1) + '}';
	}

	public void toDialect(CodeWriter writer) {
		writer.append('{');
		if(this.size()>0) {
			for(DictEntry entry : this) {
				entry.toDialect(writer);
				writer.append(", ");
			}
			writer.trimLast(2);
		}
		writer.append('}');
	}
	
}

File: Java-CodeServer/src/main/resources/js/prompto/literal/DictEntryList.js
function DictEntryList(entries, entry) {
	this.items = entries || [];
	entry = entry || null;
	if(entry!==null) {
		this.items.push(entry);
	}
	return this;
}

DictEntryList.prototype.toDialect = function(writer) {
    writer.append('{');
    if(this.items.length>0) {
        for(var i=0;i<this.items.length;i++) {
            this.items[i].toDialect(writer);
            writer.append(", ");
        }
        writer.trimLast(2);
    }
    writer.append('}');
};

DictEntryList.prototype.toString = function() {
	return "{" + this.items.join(", ") + "}";
};

DictEntryList.prototype.add = function(entry) {
	this.items.push(entry);
};

exports.DictEntryList = DictEntryList;